model/src/main/java/com/example/javaopencommerce/Amount.java

```diff
@@ -25,7 +27,29 @@ public BigDecimal asDecimal() {
     return valueOf(this.value);
   }
 
+  public Amount plus(Amount amount) {
+    return new Amount(value + amount.getValue());
+  }
+
+  public Amount minusOne() {
+    int newAmount = value == 0 ? 0 : value - 1;
+    return new Amount(newAmount);
+  }
+
+  public boolean isLessThan(Amount amount) {
+    return value < amount.getValue();
+  }
+
+  public boolean isZero() {
+    return value == 0;

```

Comment:
It's safer to change to `<=0`, could someday potentially safe from some bug.

```suggestion
    return value <= 0;

```